libs/components/packages/src/schematics/migrations/update-7/fix-mixed-case-imports.ts

```diff
@@ -0,0 +1,142 @@
+import { Path, dirname, join, split } from '@angular-devkit/core';
+import { Rule, Tree } from '@angular-devkit/schematics';
+import * as ts from '@schematics/angular/third_party/github.com/Microsoft/TypeScript/lib/typescript';
+import * as path from 'path';
+
+import { getWorkspace } from '../../utility/workspace';
+
+function updateTypescriptImports(
+  filePath: string,
+  tree: Tree
+): void {
+  const fileContent = tree.read(filePath)?.toString();
+  if (!fileContent) {
+    return;
+  }
+  const source = ts.createSourceFile(
+    filePath,
+    fileContent,
+    ts.ScriptTarget.Latest,
+    true
+  );
+  // Get all import declarations
+  const importDeclarations = source.statements.filter(
+    (statement) => statement.kind === ts.SyntaxKind.ImportDeclaration
+  ) as ts.ImportDeclaration[];
+  // Just local import targets
+  const localImportDeclarations = importDeclarations.filter(
+    (importDeclaration) => {
+      const importPath = importDeclaration.moduleSpecifier.getText(source);
+      const importPathWithoutQuotes = importPath.substring(
+        1,
+        importPath.length - 1
+      );
+      return importPathWithoutQuotes.startsWith('.');
+    }
+  );
+  const transformers: ts.TransformerFactory<ts.SourceFile>[] = [];
+  localImportDeclarations.forEach((importDeclaration) => {
+    const importPath = importDeclaration.moduleSpecifier.getText(source);
+    const importPathWithoutQuotes = importPath.substring(
+      1,
+      importPath.length - 1
+    );
+    const absoluteImportPath = path.join(path.dirname(filePath), `${importPathWithoutQuotes}.ts`.replace(/\\/g, '/'));
+    const fileEntry = tree.get(absoluteImportPath);
+    if (fileEntry && fileEntry.path === absoluteImportPath && !importPathWithoutQuotes.includes('\\')) {
+      // File exists. Do nothing.
+      return;
+    }
+    const pathFragments = split(<Path>absoluteImportPath).slice(1);
```

Comment:
This is the line causing Prettier to fail. You'll need to do `absoluteImportPath as Path` instead.